components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -259,7 +259,7 @@ export class PrebuildManager {
             } else {
                 span.setTag("starting", true);
                 const projectEnvVars = await projectEnvVarsPromise;
-                await this.workspaceStarter.startWorkspace({ span }, workspace, user, [], projectEnvVars, {
+                await this.workspaceStarter.startWorkspace({ span }, workspace, user, project, [], projectEnvVars, {
```

Comment:
I'm adding the optional project to the call graph of workspace starts here, so that we fetch the project once and pass it down instead of repeatedly asking the DB.

Reply:
It's not only because of reducing db queries but also a principle of good software design to explicitly pass what is needed as an argument rather than fetching data internally from a global state (the DB). I.e. more functional and easier to understand when you see the signature.

Reply:
> also a principle of good software design to explicitly pass what is needed as an argument rather than fetching data internally from a global state (the DB)

That is a bit too vague and general for my taste to count as an argument, honestly. Then we'd need to talk service/layer boundaries first IMO, and sometimes there are good arguments against it as well.

But let's not hold up this PR any longer, I think the change is good overall. If we disagree in style we should settle that outside of PRs. :lotus_position: